core-web/libs/block-editor/src/lib/nodes/contentlet-block/contentlet-block.node.ts

```diff
@@ -39,17 +39,35 @@ export const ContentletBlock = (injector: Injector): Node<ContentletBlockOptions
         },
 
         renderHTML({ HTMLAttributes }): DOMOutputSpec {
-            const { data } = HTMLAttributes;
-            const img = data.hasTitleImage ? ['img', { src: data.image }] : ['span', {}];
-
-            return [
-                'div',
-                { 'data-dotCMS-contentlet': 'true' },
-                ['h3', data.title],
-                ['div', data.identifier],
-                img,
-                ['div', {}, data.language]
+            type ContentletData = {
+                title?: string;
+                identifier?: string;
+                language?: string | number;
+                languageId?: string | number;
+                hasTitleImage?: boolean;
+                image?: string;
+            };
+
+            const rawData: ContentletData =
+                (HTMLAttributes as { data?: ContentletData })?.data ?? {};
+
+            const titleText = rawData.title ?? '';
+            const identifierText = rawData.identifier ?? '';
+            const languageText = rawData.language ?? rawData.languageId ?? '';
+            const hasImage = Boolean(rawData.hasTitleImage && rawData.image);
+
+            const children: DOMOutputSpec[] = [
+                ['h3', String(titleText)],
+                ['div', String(identifierText)],
+                ['div', {}, String(languageText)]
             ];
+
+            if (hasImage) {
+                // Insert image after identifier block
+                children.splice(2, 0, ['img', { src: String(rawData.image) }]);
```

Comment:
`titleImage` value will point you to the correct property image and most of the times, that field does not contain a correct URL, you have to build it.

Something like this can work:

`/dA/<contentlet-inode>/rawData.titleImage/<any-image-api-filter-needed>`